### DISCUSSION TITLE
Dynamically creating table and data model

### DISCUSSION THREAD

**Discussion by Muzzamil1** [prisma/prisma#4833] [Q&A] [answered]
Hi all!
Is it possible to generate the data model dynamically? 
My use case is that the frontend user should be able to add, modify and delete tables in the database and work with them afterward. As far as I know, the schema is saved in the Prisma data model file, which should be deployed after changes in the schema. Is there a way to deploy changes in the schema directly?

**Comment by nikolasburk** [comment, marked as answer]
Hey @Muzzamil1, if I understand your question correctly, you're looking for a way to _programmatically_ migrate the database schema. Ist that correct?

If so, then this is currently not natively supported by Prisma Migrate unfortunately, we already have an open issue for this [here](https://github.com/prisma/prisma/issues/4703). As a workaround, you can consider invoking the `prisma migrate` commands from Node.js.

Also, this library might be helpful for you: https://github.com/amplication/prisma-schema-dsl

Let me know if that helps or if you have any further questions, always happy to help!

**Reply by Muzzamil1** [reply]
Thanks @nikolasburk for answering!
Yeah, you are right. I'm looking for a way to programmatically migrate the database schema. It would be great to have this feature built-in 
 in Prisma. 
Thanks, will look into both solutions that you provided.

**Reply by vineyrawat** [reply]
@Muzzamil1 Did you found any way for doing this cause I'm also facing this problem. It would be great if you share. Thanks in advance.

**Reply by Muzzamil1** [reply]
Unfortunately No

**Reply by Muzzamil1** [reply]
Please don't mention it.

Please share the solution here if you found a way. That will be very helpful

**Reply by Malenconiaprincep** [reply]
@nikolasburk may I ask when it is planned to be developed? This is not a good fit for low-code development right now.

**Comment by TejasThombare20** [comment]
> Hi all! Is it possible to generate the data model dynamically?  My use case is that the frontend user should be able to add, modify and delete tables in the database and work with them afterward. As far as I know, the schema is saved in the Prisma data model file, which should be deployed after changes in the schema. Is there a way to deploy changes in the schema directly?

@Muzzamil1  have you got the solutions for your case

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
